ktor-network/common/test/io/ktor/network/sockets/tests/TCPSocketTest.kt

```diff
@@ -169,19 +169,22 @@ class TCPSocketTest {
     fun testReadFromCancelledSocket() = testSockets { selector ->
         val tcp = aSocket(selector).tcp()
         tcp.bind().use { server ->
-            val serverConnection = async {
+            val serverConnectionPromise = async {
                 server.accept()
             }
 
             val client: Socket = tcp.connect("127.0.0.1", server.port)
             val readChannel = client.openReadChannel()
-            serverConnection.await()
+            val serverConnection = serverConnectionPromise.await()
 
             client.cancel()
 
             assertFailsWith<CancellationException> {
                 readChannel.readByte()
             }
+
+            client.close()
+            serverConnection.close()
```

Comment:
Needed to add these close calls to prevent `jsNodeTest` task from getting stuck. Tests were all green but the gradle task just got stuck. Reproducible locally and on CI (https://ktor.teamcity.com/buildConfiguration/Ktor_KtorMatrixJavaScriptChromeNodeJs/316684). Is this a Kotlin bug?